llpc/tool/llpcCompilationUtils.cpp

```diff
@@ -76,7 +76,9 @@
 #include "llvm/Support/raw_ostream.h"
 
 #ifndef LLPC_ENABLE_SPIRV_OPT
```

Comment:
What does spirv-opt have to do with this?

Reply:
I didn't change it, what I have changed is below.
LLPC_ENABLE_SPIRV_OPT is not currently used.

Reply:
Even if you did not add that line you chose how to combine your change with that line.  Why did you not do

```
#if !defined(LLPC_ENABLE_SPIRV_OPT) || !defined(LLPC_ENABLE_SPIRV_OPT)
#define SPVGEN_STATIC_LIB
#endif
```

The names of the macros are very confusing.  You say if that if you are not using static spvgen, then you define spvgen_static_lib.  That seems backwards.

You might want to have a comment that explains the reason for making these decisions.  Also, if the macros need to change name, then we should do that.

Reply:
> The names of the macros are very confusing. You say if that if you are not using static spvgen, then you define spvgen_static_lib. That seems backwards.
> 
Yes, I agree with you. this name is backwards, but it is defined in `spvgen`, other projects are using it, I can't change it.





> The other way of combining these two options is to have the cmake file define `LLPC_USE_STATIC_SPVGEN` if `LLPC_ENABLE_SPIRV_OPT` is defined.
> 
> > LLPC_ENABLE_SPIRV_OPT is not currently used.
> 
> As long as it is in the code, we should be thinking carefully how all of the different options should interact. We need to know the reason that SPVGEN_STATIC_LIB is set when LLPC_ENABLE_SPIRV_OPT is not defined. If it is because SPVGEN_STATIC_LIB must be set, then you may have broken something because it sometimes won't be set. If it is that when LLPC_ENABLE_SPIRV_OPT is set, then SPVGEN_STATIC_LIB cannot be set, then original code is wrong because SPVGEN_STATIC_LIB could be set somewhere else.

Yes, Here is also very confusing. My understanding is that we need `spvgen` when `LLPC_ENABLE_SPIRV_OPT' is defined, then we could select static or shared library; if  `LLPC_ENABLE_SPIRV_OPT' is undefined, we don't need `spvgen`, which means `spvgen.h` should be gated into macros.
However, I am worried that I've misunderstood, so I didn't make changes. I only make sure `amdllpc` works.

Reply:
Okay.